tests/testthat/test-sort_linter.R

```diff
@@ -0,0 +1,21 @@
+test_that("sort_linter skips allowed usages", {
+  expect_lint("order(y)", NULL, sort_linter())
+
+  expect_lint("y[order(x)]", NULL, sort_linter())
+
+  # If another function is intercalated, don't fail
+  expect_lint("x[c(order(x))]", NULL, sort_linter())
+})
+
+
+test_that("sort_linter blocks simple disallowed usages", {
+  lint_message <- rex::rex("sort(x) is better than x[order(x)].")
+
+  expect_lint("x[order(x)]", lint_message, sort_linter())
+
+  # Works with extra args in order()
+  expect_lint("x[order(x, decreasing = TRUE)]", lint_message, sort_linter())
```

Comment:
This is a good point too -- our recommendation should more broadly endeavour to match other arguments in the shared signatures of `sort()` and `order()`, namely `method`, `decreasing`, and `na.last`.

Reply:
How should I go about this? I tried to keep using the xmlparsedata output but I'm not sure it's the best way:

```r
extra_args <- xml2::xml_text(xml2::xml_find_all(bad_expr, "//SYMBOL_SUB"))
extra_arg_values <- xml2::xml_text(xml2::xml_find_all(bad_expr, "//SYMBOL_SUB/following-sibling::expr"))
```

Two problems:

- this wouldn't work with unnamed arguments as far as I can tell
- since the arg name and the arg value are captured separately, I would have to reconstruct the call with `paste()` or `glue()`

An alternative would be to start from `xml2::xml_text(bad_expr)`. It returns something like `order(x,decreasing=TRUE,method='radix')` with the current XPath. I would then have to extract the arguments and values so it seems to defeat to purpose of using xmlparsedata earlier.

Reply:
I think I'm not clear on what you're asking exactly:

- should I only return a warning when the `order()` call includes one of `method`, `decreasing`, `na.last`?
- should I customize the warning message to match the extra arguments in the call? (kind of like what you describe in https://github.com/r-lib/lintr/pull/1528#discussion_r973968141 but for all arguments)

Reply:
Thanks for your work here Hugo. I our north star is we need to recommend a correct alternative in the lint message.

At a minimum, I think that means being careful about `na.last`, since `sort()` and `order()` differ on their defaults.

(I don't have R in front of me, but I seem to recall all the other arguments share the same default, so it's not as important to capture that in the recommendation -- doing so can be an aspirational follow-up feature).

More specifically addressing your questions
 - No, I think the logic to recommend `sort()` when user is doing `x[order(x)]` applies regardless of the other arguments. The only caveat is that they need to be careful about `na.last` when making the switch.
 - If you can work out how to do so, that's great. It's much friendlier to see a lint message that closely matches your actual code. However, it's not required for approving the PR, so don't bend over backward to accomplish it, we can put it in the TODO queue for future work.

Reply:
Thanks for your answer. It clears things up for me.

It should be implemented in [`28a3b51` (#1528)](https://github.com/r-lib/lintr/pull/1528/commits/28a3b5143b39ac5d32cab58e461b7187b2a87ff2).